Consider error-deprecated `ClassName(String)` overload

Sometimes I accidentally forget to split the simple name and write `ClassName("com.example.Foo")`. This fails at runtime.
```
Caused by: java.lang.IllegalArgumentException: simpleNames must not be empty
        at com.squareup.kotlinpoet.ClassName.<init>(TypeName.kt:323)
```

It would be nice if it failed at compile-time. This can be done with something like
```kotlin
@Deprecated("Simple names must not be empty", level = ERROR)
fun ClassName(value: String): Nothing = throw IAE(..)
```

--- a/kotlinpoet/src/main/java/com/squareup/kotlinpoet/ClassName.kt
+++ b/kotlinpoet/src/main/java/com/squareup/kotlinpoet/ClassName.kt
@@ -40,6 +40,13 @@ public class ClassName internal constructor(
   public constructor(packageName: String, simpleName: String, vararg simpleNames: String) :
     this(listOf(packageName, simpleName, *simpleNames))
 
+  @Deprecated(
+    "Simple names must not be empty. Did you forget an argument?",
+    level = DeprecationLevel.ERROR,
+    replaceWith = ReplaceWith("ClassName(packageName, TODO())"),
+  )
+  public constructor(packageName: String) : this(packageName, listOf())
+
   /**
    * Returns a class name created from the given parts. For example, calling this with package name
    * `"java.util"` and simple names `"Map"`, `"Entry"` yields `Map.Entry`.

--- a/kotlinpoet/src/test/java/com/squareup/kotlinpoet/ClassNameTest.kt
+++ b/kotlinpoet/src/test/java/com/squareup/kotlinpoet/ClassNameTest.kt
@@ -139,6 +139,7 @@ class ClassNameTest {
     // }
   }
 
+  @Suppress("DEPRECATION_ERROR") // Ensure still throws in case called from Java.
   @Test fun fromEmptySimpleName() {
     assertThrows<IllegalArgumentException> {
       ClassName("foo" /* no simple name */)

